### PR TITLE
Branch documentations

### PR DESCRIPTION
**Pull request by yuhaochua**
Add instructions for manual testing to developer guide.

Modified sequence diagram and added to glossary in developer guide.

Discovered a bug which causes application to crash when using CheckCommand incorrectly and so I fixed it within this pull request as well.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 - [Design and Implementation](#design-and-implementation)
   - [Architecture](#architecture)
   - [Parser](#parser)
-  - [Update feature](#update-feature)
+  - [Command classes](#command-classes)
```
Comment (shunyao643) [inline docs/DeveloperGuide.md:7]: Thanks for catching this one! I missed this out earlier.

File: src/test/java/seedu/parser/ParserTest.java [hunk 2]
```diff
@@ -337,13 +337,6 @@ void extractArguments_wrongArgTypesUsed_exceptionThrown() {
         assertEquals(IncompleteCommandException.NO_PARAMETERS_FOUND, exception.getMessage());
     }
 
-    @Test
-    void parseCommand_listEnumTypeConvertedToUpper_success() {
-        Command testCommand = parser.parseCommand("list spEAker");
-        Command expectedCommand = new ListCommand(new ArrayList<>(Collections.singleton("SPEAKER")));
-        assertEquals(expectedCommand, testCommand);
-    }
-
```
Comment (shunyao643) [inline src/test/java/seedu/parser/ParserTest.java:346]: Cheers for helping me remove this!

File: src/main/java/seedu/command/Command.java [hunk 3]
```diff
@@ -13,11 +14,13 @@
     public String usageReminder;
     protected EquipmentManager equipmentManager;
     protected Storage storage;
-    public static final String INCORRECT_ENUM_TYPE = "Wrong type of equipment. The allowed types are: "
-            + "MICROPHONE, SPEAKER, STAND, CABLE";
+    public static final String INCORRECT_ENUM_TYPE = "Equipment Type needs to be one of:" + System.lineSeparator()
+            + EquipmentType.getAllTypes();
```
Comment (shunyao643) [inline src/main/java/seedu/command/Command.java:18]: Great idea with this!

**Review by shunyao643 — APPROVED** [review]
Looks good to merge, thanks for putting this together!

**Review by Bryan-BC — APPROVED** [review]
LGTM. btw, I think for the docs/team/yuhao.md , the name of the file needs to match your github cuz i got email about missing file a few days ago

**Pull request merged**